common/src/main/java/org/keycloak/common/Profile.java

```diff
@@ -89,6 +89,13 @@ public Profile(PropertyResolver resolver) {
                     break;
             }
         }
+
+        if (Boolean.TRUE.equals(config.getConfig(Feature.ADMIN2)) || Boolean.TRUE.equals(config.getConfig(Feature.ADMIN))) {
+            // re-enable admin-api if admin-console feature is activated.
+            if (disabledFeatures.remove(Feature.ADMIN_API)) {
+                logger.warnf("Enabled disabled feature %s because it is required by the enabled admin-console.", Feature.ADMIN_API);
+            }
+        }
```

Comment:
I'm not sure about this one, maybe it would be better to just throw a RuntimeException stating admin console requires admin apis to be enabled?

Reply:
Might also be good to instead of hardcoding this add some support for a feature to require other features. That is a refactoring we could do after this PR though.

Reply:
How about:
```java
        if (Boolean.TRUE.equals(config.getConfig(Feature.ADMIN2)) || Boolean.TRUE.equals(config.getConfig(Feature.ADMIN))) {
            if (disabledFeatures.contains(Feature.ADMIN_API)) {
                throw new RuntimeException(String.format("Invalid value for feature: %s needs to be enabled because it is required by %s.",
                        Feature.ADMIN_API, Arrays.asList(Feature.ADMIN, Feature.ADMIN2)));
            }
        }
```

Reply:
That works